src/libsystemd/sd-device/sd-device.c

```diff
@@ -245,6 +246,28 @@ _public_ int sd_device_new_from_devnum(sd_device **ret, char type, dev_t devnum)
         return sd_device_new_from_syspath(ret, syspath);
 }
 
+_public_ int sd_device_new_from_ifname(sd_device **ret, const char *ifname) {
+        const char *syspath;
+
+        assert_return(ret, -EINVAL);
+        assert_return(ifname, -EINVAL);
```

Comment:
maybe validate the input as an ifname with ifname_valid(). i.e. it kinda makes me more comfortable if we concatenate the name to a path prefix only after validating it as valid ifname, so that nothing weird can happen regarding input strings that include "../" and suchlike

Reply:
Oh, indeed. Fixed.